pkg/lib/indexer/indexer.go

```diff
@@ -513,38 +523,72 @@ func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
 	}
 
 	var errs []error
+	var g errgroup.Group
+	var mu = &sync.Mutex{}
+
+	concurrentGoroutines := make(chan struct{}, concurrencyLimitForExport)
+
 	for _, bundleImage := range bundles {
-		// try to name the folder
-		folderName, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
-		if err != nil {
+		bundleImage := bundleImage // to ensure every go routine gets a it's own copy of closure variable
+
+		g.Go(func() error {
+			concurrentGoroutines <- struct{}{}
+			defer func() { <-concurrentGoroutines }()
+
+			// get the bundle version to be used in naming the folder
+			bundleVersion, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
```

Comment:
Do we need to make a separate transaction to the database for each package? Can't we just get this all at once when we do the initial package query above?

Reply:
modified the code